File: GFG/Array/Day4/KadanesAlgorithm.java
public class KadanesAlgorithm {
    public int maxSubarraySum(int[] arr) {
        // Code here
        int cursum,maxsum;
        cursum=0;
        maxsum=arr[0];
        
        for(int n:arr){
            if(cursum<0){
                cursum=0;
            }
            cursum+=n;
            maxsum=Math.max(cursum,maxsum);
        }
        
        return maxsum;
    }
}

File: GFG/Array/Day4/KadanesAlgorithm.py
class Solution:
    def maxSubarraySum(self, arr):
        # Code here
        cursum=0
        maxsum=arr[0]
        
        for num in arr:
            if(cursum<0):
                cursum=0
            cursum+=num
            maxsum=max(cursum,maxsum)
            
        return maxsum